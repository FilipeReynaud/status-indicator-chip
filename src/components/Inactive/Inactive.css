.inactive-div {
	width: 80px;
	height: 25px;
	background-color: #fdeeee;
	border-radius: 50px;
	position: relative;
	margin: auto;
}

.inactive-circle {
	width: 10px;
	height: 10px;
	background-color: #ee5f5b;
	border-radius: 100%;
	position: absolute;
	margin-top: 7.5px;
	margin-left: 7.5px;
}

.inactive-text {
	color: #ee5f5b;
	font-family: "Roboto", sans-serif;
	font-weight: bold;
	font-size: 11px;
	position: absolute;
	right: 5px;
	top: 4px;
}
