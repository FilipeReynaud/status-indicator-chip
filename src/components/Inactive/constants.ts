// Styles
const COLOR = "#ee5f5b";
const BACKGROUND_COLOR = "#fdeeee";

// Defaults
const LABEL = "Inactive";
const SIZE = "md";

export { COLOR, SIZE, BACKGROUND_COLOR, LABEL };
