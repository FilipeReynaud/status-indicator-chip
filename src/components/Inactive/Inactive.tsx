import React from "react";

// Components
import BaseChip from "../BaseChip";

// Constants
import * as constants from "./constants";

// Types
export interface InactiveProps {
	label?: string;
	size?: "sm" | "md" | "lg";
}

const Inactive = (props: InactiveProps) => (
	<BaseChip
		fontColor={constants.COLOR}
		backgroundColor={constants.BACKGROUND_COLOR}
		label={props.label ?? constants.LABEL}
		size={props.size ?? constants.SIZE}
		withBlinkAnimation={false}
	/>
);

export default Inactive;
