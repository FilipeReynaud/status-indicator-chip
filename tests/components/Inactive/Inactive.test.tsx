import React from "react";
import { render, screen } from "@testing-library/react";

// Components
import Inactive from "../../../src/components/Inactive/Inactive";

describe("Inactive", () => {
	it("should render properly", () => {
		// Act
		const { container } = render(<Inactive />);

		// Assert
		expect(container).toMatchSnapshot();
	});

	it("should have 'Inactive' label", async () => {
		// Arrange
		render(<Inactive />);

		// Assert
		expect(await screen.findByText("Inactive")).toBeTruthy();
	});

	it("should be able to set another label", async () => {
		// Arrange
		const defaultLabel = "Inactive";
		const testLabel = "Dispatched";

		render(<Inactive label={testLabel} />);

		// Assert
		expect(await screen.queryByText(defaultLabel)).toBeNull();
		expect(await screen.findByText(testLabel)).toBeTruthy();
	});
});
