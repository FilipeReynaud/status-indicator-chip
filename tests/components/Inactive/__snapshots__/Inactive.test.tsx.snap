// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inactive should render properly 1`] = `
<div>
  <table
    class="status-indicator-table"
    data-test="status-indicator-table"
    style="padding: 0px 7px 0px 5px; background-color: rgb(253, 238, 238);"
  >
    <tbody>
      <tr>
        <td>
          <div
            class="status-indicator-circle"
            data-test="status-indicator-circle"
            style="width: 10px; height: 10px; background-color: rgb(238, 95, 91);"
          />
        </td>
        <td>
          <span
            class="status-indicator-label"
            data-test="status-indicator-label"
            style="color: rgb(238, 95, 91); font-size: 11px;"
          >
            Inactive
          </span>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
